lib/rules/no-duplicate-case.js

```diff
@@ -0,0 +1,57 @@
+/**
+ * @fileoverview Rule to disallow a duplicate case label.
+ * @author Dieter Oberkofler
+ * @copyright 2015 Dieter Oberkofler. All rights reserved.
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+
+    /**
+     * Get a hash value for the test
+     * @param {ASTNode} test The "test" node.
+     * @returns {string} A hash value for the test.
+     * @private
+     */
+    function getHash(test) {
+        if (test.type === "Literal") {
+            return test.type + typeof test.value + test.value;
+        } else if (test.type === "Identifier") {
+            return test.type + typeof test.name + test.name;
+        }
+    }
+
+    return {
+
+        "SwitchStatement": function(node) {
+
+            var caseCount = node.cases.length,
+                hashTable = {},
+                caseNode,
+                hashValue,
+                i;
+
+            for (i = 0; i < caseCount - 1; i++) {
```

Comment:
Wow! That's a fancy way of doing this and I'm starting to better understand the concepts of how and why traversing the nodes.

I'm interested in better understanding, why you prefer this approach and can think of:
1) Standardization
2) Performance (is this faster than looping and if so why)
3) Code size/complexity (it's a little shorter but also seems a little harder to understand)